artemis-server/src/main/resources/schema/artemis-configuration.xsd

```diff
@@ -675,6 +675,82 @@
             </xsd:annotation>
          </xsd:element>
 
+         <xsd:element name="journal-retention" maxOccurs="1" minOccurs="0">
+            <xsd:annotation>
+               <xsd:documentation>
+                  Configuration about the journal retention configuration.
+               </xsd:documentation>
+            </xsd:annotation>
+            <xsd:complexType>
+               <xsd:attribute name="unit" use="required">
+                  <xsd:annotation>
+                     <xsd:documentation>
+                        This configures the period type to use on limit. By default it is DAYS.
+                     </xsd:documentation>
+                  </xsd:annotation>
+                  <xsd:simpleType>
+                     <xsd:restriction base="xsd:string">
```

Comment:
Same comment as below what if someone wants to limit just on size

Reply:
Also why seperate the unit and the period why not as a single attribute string like for storage where its value and scaler

E.g. 2040594944ms    360s  4h 7d   and if no unit default to ms

Reply:
it wouldn't be as clear. the XSD is pretty standard.. if you edit the XML with an editor you get auto-completion and self documented. As if you use the schema you use on bytes it would be free form... more error prone.